ui/js/graphs.js

```diff
@@ -769,6 +786,73 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
     $scope.testsToAdd = []; // tests in the "tests to add" list
     $scope.selectedTestsToAdd = []; // tests in the "to add" test list that have been selected by the user
 
+    var relateSeries = options.relatedSeries;
+    // if we choose to add all related platforms, firstly we need
+    // to obtain all the platform of this project, then pick out the
+    // specific test by platform and project.
+    var addRelatePlatform = function(relateSeries) {
+        PhSeries.getAllSeries(relateSeries.projectName,
```

Comment:
Incidentally, once you add the platform list endpoint in bug 1189519, you could make this much faster.